FF129 MediaCapabilities.decodingInfo() with keySystemAccept arg (#34836)
mdn/content: files/en-us/web/api/mediacapabilities/decodinginfo/index.md

        - `samplerate`
          - : The number of audio samples making up one second of the audio file.

>     - `keySystemConfiguration` {{optional_inline}}

Author: This is the encrypted content configuration to test (i.e the new bit).

>       - : Object specifying the key system configuration for encrypted media.
> 
>         If specified, the [`type`](#type) must be `media-source` or `file` (not `webrtc`).
>         This has the following properties: <!-- MediaCapabilitiesKeySystemConfiguration in the spec -->
> 
>         - `keySystem`
> 
>           - : A string identifying the media key system.
>             For example `com.example.somesystem` or `org.w3.clearkey`.
> 
>         - `initDataType` {{optional_inline}}
> 
>           - : A string indicating the data type name the initialization data format, such as `"cenc"`, `"keyids"` and `"webm"`.
>             Allowed names are defined in the [Encrypted Media Extensions Initialization Data Format Registry](https://www.w3.org/TR/eme-initdata-registry/).
>             Note that [`Navigator.requestMediaKeySystemAccess()`](/en-US/docs/Web/API/Navigator/requestMediaKeySystemAccess) takes an array of data type names in the `supportedConfigurations.initDataTypes` argument.

Author: ```suggestion
      - : Object specifying the key system configuration for encrypted media.
      
        Note that [`Navigator.requestMediaKeySystemAccess()`](/en-US/docs/Web/API/Navigator/requestMediaKeySystemAccess) takes arrays some of the same data types in its `supportedConfigurations` argument.

        If specified, the [`type`](#type) must be `media-source` or `file` (not `webrtc`).
        This has the following properties: <!-- MediaCapabilitiesKeySystemConfiguration in the spec -->

        - `keySystem`

          - : A string identifying the media key system.
            For example `com.example.somesystem` or `org.w3.clearkey`.

        - `initDataType` {{optional_inline}}

          - : A string indicating the data type name the initialization data format, such as `"cenc"`, `"keyids"` and `"webm"`.
            Allowed names are defined in the [Encrypted Media Extensions Initialization Data Format Registry](https://www.w3.org/TR/eme-initdata-registry/).
```

>           - : A string indicating the data type name the initialization data format, such as `"cenc"`, `"keyids"` and `"webm"`.
>             This is a single value from the array defined in the [`supportedConfigurations.initDataTypes`](/en-US/docs/Web/API/Navigator/requestMediaKeySystemAccess#initdatatypes) option passed to `Navigator.requestMediaKeySystemAccess()`.
>             Allowed names are defined in the [Encrypted Media Extensions Initialization Data Format Registry](https://www.w3.org/TR/eme-initdata-registry/).

Author: ```suggestion
          - : A string indicating the data type name the initialization data format, such as `"cenc"`, `"keyids"` and `"webm"`.
            Allowed names are defined in the [Encrypted Media Extensions Initialization Data Format Registry](https://www.w3.org/TR/eme-initdata-registry/).
            Note that [`Navigator.requestMediaKeySystemAccess()`](/en-US/docs/Web/API/Navigator/requestMediaKeySystemAccess) takes an array of data type names in the `supportedConfigurations.initDataTypes` argument.
```

                This value should be set by an application (it defaults to `null`, indicating that any encryption scheme may be used).
            - `robustness`
              - : The robustness level associated with the content type.
                The empty string indicates that any ability to decrypt and decode the content type is acceptable.

### Return value

A {{jsxref('Promise')}} fulfilling with an object containing the following attributes:

- `supported`
  - : `true` if the media content can be decoded at all. Otherwise, it is `false`.
- `smooth`
  - : `true` if playback of the media can be played at the frame rate specified by the configuration without needing to drop frames. Otherwise it is `false`.
- `powerEfficient`
  - : `true` if playback of the media will be power efficient. Otherwise, it is `false`.
- `keySystemAccess`
  - : A {{domxref("MediaKeySystemAccess")}} that can be used to create a {{domxref("MediaKeys")}} object to setup encrypted playback, or `null` if decoding is not supported using the supplied configuration.

Browsers will report a supported media configuration as `smooth` and `powerEfficient` until stats on this device have been recorded.

> All supported audio codecs report `powerEfficient` as true.

Author: I can't see this in the specification, but it makes sense, and preexists this change.

};
```

> ### Getting decoding information for encrypted media

Author: This is from the explainer, and should be good. I find this stuff hard to test, and I'm worried I would spend a whole month just on this if I tried.
